### DISCUSSION TITLE
Processing dMRI (dicom) using DIPY

### DISCUSSION THREAD

**Discussion by hjara21** [dipy/dipy#2624] [Preprocessing]
I modified your code (attached) to read locally a set of dicom dMRI (7 directions). The code uses dcm2nifti to generate niftis (separate for b0 and b1000) storing them in a folder (nifti_directory) and when it tries to read them for dMRI processing, it gives the following error.

Traceback (most recent call last):
  File "e:/00 & D/00 & Python Development Anac/01e & PFG_TripleTSE/reconst_dti.py", line 93, in <module>
    hardi_fname, hardi_bval_fname, hardi_bvec_fname = get_fnames('nifti_directory')
TypeError: cannot unpack non-iterable NoneType object

**Comment by hjara21** [comment]
Sorry I could not attach py file so I zipped it
[reconst_dti.zip](https://github.com/dipy/dipy/files/9124624/reconst_dti.zip)
.

**Comment by hjara21** [comment]
The ultimate goal is to be able to load dicom files, convert these to nifti, DTI processing, and save the DTI (MD, FA, etc) maps.

**Comment by pjsjongsung** [comment]
get_fnames function does not support a custom directory. It is a function to fetch data already in the dipy package. 
A simple solution would be getting rid of 
```
hardi_fname, hardi_bval_fname, hardi_bvec_fname = get_fnames('nifti_directory')
print(str(hardi_fname))
data, affine = load_nifti(hardi_fname)

bvals, bvecs = read_bvals_bvecs(hardi_bval_fname, hardi_bvec_fname)
gtab = gradient_table(bvals, bvecs)
```

altogether and replace them with your own nifti file and bval and bvec files.

ex)
```
data, affine = load_nifti(os.path.join('nifti_directory', nifti_file))
bvals, bvecs = read_bvals_bvecs(file_with_bvalues, file_with_bvectors)
gtab = gradient_table(bvals, bvecs)
```

and then the rest of the code.

You should have some files like ~.bval ~.bvec which can replace the file_with_bvalues and file_with_bvectors above.

**Comment by hjara21** [comment]
Thank you so much! I will work on it.
Best,

**Comment by hjara21** [comment]
Thank you so much for your help yesterday. It worked.
Now, I am encountering a problem
Traceback (most recent call last):
  File "e:/00 & D/00 & Python Development Anac/01e & PFG_TripleTSE/reconst_dti.py", line 82, in <module>
    gtab = gradient_table(bvals, bvecs)
  File "C:\Users\Hernan\Anaconda3\lib\site-packages\dipy\core\gradients.py", line 599, in gradient_table
    return gradient_table_from_bvals_bvecs(bvals, bvecs, big_delta=big_delta,
  File "C:\Users\Hernan\Anaconda3\lib\site-packages\dipy\core\gradients.py", line 309, in gradient_table_from_bvals_bvecs
    raise ValueError("The vectors in bvecs should be unit (The tolerance "
ValueError: The vectors in bvecs should be unit (The tolerance can be modified as an input parameter
[bvals.zip](https://github.com/dipy/dipy/files/9126112/bvals.zip)
)

**Comment by hjara21** [comment]
Also, bvals and bvecs are respectively:
[   0. 1000. 1000. 1000. 1000. 1000. 1000. 1000.]
[[ 0.        -0.         0.       ]
 [-0.317446   0.646749  -0.693501 ]
 [-0.684334   0.350016   0.639669 ]
 [ 0.656441   0.677647   0.331481 ]
 [-0.708365   0.704819  -0.0380652]
 [-0.019723   0.726667   0.686707 ]
 [ 0.688642   0.0218481  0.724772 ]
 [ 0.        -0.         0.       ]]

**Comment by pjsjongsung** [comment]
I'm not sure what your data looks like, but the gradient_table function requires a bvecs to be a unit vector.(The ones that are not b0)

How to temper with your data is out of my knowledge, but one simple thing could be normalizing your bvecs to be a unit vector and removing the last image that has a 0 vector.
